Docker topic (#1507)
enkidevs/curriculum: docker/fundamentals/storage/docker-bind-mounts.md

---
author: nem035
aspects:
  - introduction
  - workout
  - deep
type: normal
category: must-know
links:
- '[Docker: use bind mounts](https://docs.docker.com/storage/bind-mounts/){documentation}'
---
# Bind Mounts
---

## Content

A sharing (mounting) a host directory/file into a container that just looks like a local file path - container is unaware this is coming from a host. Essentially, a bind mount is just a mapping of a host file or directory to a container file or directory. Unlike volumes which are completely managed by Docker, bind mounts are dependent on the directory structure of the host machine. This means that host files overwrite any files in the container. When we use a bind mount, a file or directory on the host machine is mounted into a container.

Since bind mounts are host-specific, they cannot be part of a Dockerfile and have to be used at run-time with `docker container run`:

Similarly to volumes, a bind mount is created with the `-v` flag but we pass in a full path instead of a name.

```bash

> docker container run -v

Reviewer: ```suggestion
docker container run \
```

  -v ~/enki/local_path:/path/container mysql
```

TEXT TO USE TO EXTRACT QUESTION:

> > Bind mounts have limited functionality compared to volumes. When you use a bind mount, a file or directory on the host machine is mounted into a container. The file or directory is referenced by its full or relative path on the host machine. By contrast, when you use a volume, a new directory is created within Docker’s storage directory on the host machine, and Docker manages that directory’s contents.

Reviewer: **PQ**

A bind mount is ???.

* a file or directory on host machine directly mounted on a path into a container
* a volume on a host machine that is mounted on a path into a container
* an IP address and port bound from host machine to container
* an ftp connection abstracted and mounted on a path inside a container

___

**RQ**

Create a bind-mount when launching a container from local path `~/local/data` to `/local-data`:

```bash
docker container run \
  ??? ???:??? nginx
```

* -v
* ~/local/data
* /local-data
* -b
* --bound

---
author: nem035
aspects:
  - introduction
  - workout
  - deep
type: normal
category: must-know
links:
- '[Docker: use bind mounts](https://docs.docker.com/storage/bind-mounts/){documentation}'
---
# Bind Mounts
---

## Content

> A sharing (mounting) a host directory/file into a container that just looks like a local file path - container is unaware this is coming from a host. Essentially, a bind mount is just a mapping of a host file or directory to a container file or directory. Unlike volumes which are completely managed by Docker, bind mounts are dependent on the directory structure of the host machine. This means that host files overwrite any files in the container. When we use a bind mount, a file or directory on the host machine is mounted into a container.

Author: I meant to remove the "that" part

> A sharing (mounting) a host directory/file into a container that just looks like a local file path - container is unaware this is coming from a host. Essentially, a bind mount is just a mapping of a host file or directory to a container file or directory. Unlike volumes which are completely managed by Docker, bind mounts are dependent on the directory structure of the host machine. This means that host files overwrite any files in the container. When we use a bind mount, a file or directory on the host machine is mounted into a container.

Author: ```suggestion
A bind mount is a mapping of a host file or directory to a container file or directory. Mounting a host directory/file into a container behaves like a local file path - the container is unaware this is coming from a host. 

Unlike volumes which are completely managed by Docker, bind mounts are dependent on the directory structure of the host machine. This means that host files overwrite any files in the container.
```
